Google Generative AI: Document service for prompts consisting of text and images using Gemini Pro Vision (#30354)
home-assistant/home-assistant.github.io: source/_integrations/google_generative_ai_conversation.markdown

  description: Number of top-scored tokens to consider during generation.

{% endconfiguration_basic %}

## Services

### Service `google_generative_ai_conversation.generate_content`

Allows you to ask Gemini Pro or Gemini Pro Vision to generate content from a prompt consisting of text and optionally images.

> This service populates [Response Data](/docs/scripts/service-calls#use-templates-to-handle-response-data) with the generated content.

Reviewer: ```suggestion
This service populates [response data](/docs/scripts/service-calls#use-templates-to-handle-response-data) with the generated content.
```

| Service data attribute | Optional | Description                                    | Example             |
| ---------------------- | -------- | ---------------------------------------------- | ------------------- |
| `prompt`               | no       | The prompt for generating the content.         | Describe this image |

> | `image_filename`       | yes      | Filenames for images to include in the prompt. | /tmp/image.jpg      |

Reviewer: ```suggestion
| `image_filename`       | yes      | File names for images to include in the prompt. | /tmp/image.jpg      |
```

{% raw %}
```yaml
service: google_generative_ai_conversation.generate_content
data:
  prompt: >-

>     Very briefly describe what you see in this image from my doorbell camera.
>     Your message needs to be short to fit in a phone notification. Don't
>     descrive stationary objects or buildings.

Reviewer: ```suggestion
    Very briefly describe what you see in this image from my doorbell camera.
    Your message needs to be short to fit in a phone notification. Don't
    describe stationary objects or buildings.
```
